### ISSUE TITLE
Compatibility with S3 as Data Backend

### ISSUE THREAD

**Issue by ching-kuo** [ariselseng/camerarawpreviews#97]
Environment:
Nextcloud version: 27.0.1
Data Backend: S3

This plugin seems to be using a newer version of psr, which is incompatible with AWS PHP SDK causing all uploads to fail with ERROR message.

```
{"reqId":"Yr5genuHA4oBaunNggfE","level":3,"time":"2023-07-30T13:00:47+00:00","remoteAddr":"100.109.111.165","user":"genekuo","app":"PHP","method":"PUT","url":"/remote.php/dav/files/folder/test.file","message":"Declaration of Aws\\HashingStream::seek($offset, $whence = Aws\\SEEK_SET) must be compatible with Psr\\Http\\Message\\StreamInterface::seek(int $offset, int $whence = Psr\\Http\\Message\\SEEK_SET): void at /var/www/html/3rdparty/aws/aws-sdk-php/src/HashingStream.php#50","userAgent":"rclone/v1.62.2","version":"27.0.1.2","data":{"app":"PHP"}}
```

Disabling the plugin will rollback to the default version of psr that comes with Nextcloud which solves this issue.

I'm not a PHP developer and I'm unsure if it possible to have 2 version of library in the same environment or possible to disable one of them.

**Comment by joshtrichards** [comment]
Hit it here too: https://github.com/nextcloud/server/issues/39981

**Comment by ariselseng** [comment]
@ching-kuo @joshtrichards 
Care to test this release by installing it manually? https://github.com/ariselseng/camerarawpreviews/releases/tag/v0.8.3

**Comment by ariselseng** [comment]
@ching-kuo @joshtrichards Still waiting for someone to confirm if it works.

**Comment by ching-kuo** [comment]
@ariselseng Sorry for the delay, was on vacation for a couple weeks.
Let me look into how to install it manually, I'm running in container which will probably make it a bit tricky.

**Comment by ching-kuo** [comment]
Hi @ariselseng,

Tried 0.8.3, it works without issues. Thanks a lot!

**Comment by pantherale0** [comment]
I've just tried the linked release above and get the following error (27.1.4):

`Application serverinfo (serverinfo) is not present or has a non-compatible version with this server. Please check the apps directory.`

**Comment by pantherale0** [comment]
Good shout - didn't even think to check the permissions lol. That'll probably be the problem.

Thanks

**Comment by ariselseng** [comment]
@pantherale0 While you are at it, could you please try this new release candidate? https://github.com/ariselseng/camerarawpreviews/releases/tag/v0.8.4

I got rid of the original problem by not using the needed dependency at all.
@ching-kuo  Could you try it as well?

If it works for you guys I will make this available in the Nextcloud app store

**Comment by ching-kuo** [comment]
@ariselseng I'm currently using 0.8.3 with 27.1.4 without any issue.

**Comment by ariselseng** [comment]
@ching-kuo 0.8.4 Is almost a total rewrite as a more long term fix for this. I would appreciate a quick test of 0.8.4.
I never pushed 0.8.3 to the Nextcloud app store, so most people with S3 still have this issue.

**Comment by ching-kuo** [comment]
@ariselseng Just installed 0.8.4 with tarball. Tested viewing RAW and uploading 3.1GB file and no issue. Please let me know if there is further function you would like me to test. Thanks!

**Comment by ariselseng** [comment]
@ching-kuo @pantherale0 I have now released 0.8.4 to the Nextcloud app store. Closing this. Thank you for you testing and input.
